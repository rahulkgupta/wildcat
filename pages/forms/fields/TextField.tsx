import { TextField } from '../../util/form';

interface Props {
    field: TextField
}


const TextFieldView = (props: Props) => {

    let field = props.field
    return (
        <div>
            <label>{field.getLabel()}</label>
            <input type="text" value={field.getValue()} onChange={(event) =>{field.setValue(event.target.value);}}></input>
        </div>
    )
}

export default TextFieldView;