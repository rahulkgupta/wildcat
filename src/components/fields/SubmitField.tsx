import SubmitField from '@src/util/form/fields/submit';

interface Props {
  field: SubmitField;
  submit: Function;
}

/**
 * Renders the react view for a {@link TextField}
 * uses the various functions of {@link TextField} to handle display and updates.
 * @param props contains the {@link TextField}
 */
const SubmitFieldView = (props: Props) => {
  const field = props.field;
  return (
    <input
      type="submit"
      value={field.getValue()}
      onClick={(event) => {
        props.submit();
      }}
      data-testid={field.getId()}
    ></input>
  );
};

export default SubmitFieldView;
