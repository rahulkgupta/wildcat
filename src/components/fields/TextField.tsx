import TextField from '@src/util/form/fields/text';

interface Props {
  field: TextField;
}

/**
 * Renders the react view for a {@link TextField}
 * uses the various functions of {@link TextField} to handle display and updates.
 * @param props contains the {@link TextField}
 */
const TextFieldView = (props: Props) => {
  const { field } = props;
  return (
    <div>
      <label>{field.getLabel()}</label>
      <input
        type="text"
        value={field.getValue()}
        onChange={(event) => {
          field.setValue(event.target.value);
        }}
        data-testid={field.getId()}
      />
    </div>
  );
};

export default TextFieldView;
