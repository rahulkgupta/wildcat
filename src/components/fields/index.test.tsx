import renderer, { ReactTestRenderer } from 'react-test-renderer';
import TextField from '@src/util/form/fields/text';
import FieldView from '.';
import SubmitField from '@src/util/form/fields/submit';
import Field from '@src/util/form/fields';

class UnknownField extends Field {
  toJSON() {
    throw new Error('Method not implemented.');
  }
}

describe('FieldView', () => {
  test('renders TextFieldView', () => {
    const textFieldView: ReactTestRenderer = renderer.create(
      <FieldView
        {...{
          field: new TextField(
            '1234',
            {
              value: 'value',
              label: 'label',
            },
            jest.fn(),
          ),
          submit: jest.fn(),
        }}
      />,
    );
    expect(textFieldView.toJSON()).toMatchSnapshot();
  });

  test('renders SubmitFieldView', () => {
    const textFieldView: ReactTestRenderer = renderer.create(
      <FieldView
        {...{
          field: new SubmitField(
            '1234',
            {
              value: 'value',
              label: 'label',
            },
            jest.fn(),
          ),
          submit: jest.fn(),
        }}
      />,
    );
    expect(textFieldView.toJSON()).toMatchSnapshot();
  });

  test('renders error', () => {
    const textFieldView: ReactTestRenderer = renderer.create(
      <FieldView
        {...{
          field: new UnknownField(
            '1234',
            {
              value: 'value',
              label: 'label',
            },
            jest.fn(),
          ),
          submit: jest.fn(),
        }}
      />,
    );
    expect(textFieldView.toJSON()).toMatchSnapshot();
  });
});
