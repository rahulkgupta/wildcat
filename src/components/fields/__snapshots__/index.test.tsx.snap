// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldView renders SubmitFieldView 1`] = `
<input
  data-testid="1234"
  onClick={[Function]}
  type="submit"
  value="value"
/>
`;

exports[`FieldView renders TextFieldView 1`] = `
<div>
  <label>
    label
  </label>
  <input
    data-testid="1234"
    onChange={[Function]}
    type="text"
    value="value"
  />
</div>
`;

exports[`FieldView renders error 1`] = `
<div>
  Oops something went wrong
</div>
`;
